### PR TITLE
Serial backend modules

### PR DESCRIPTION
**Pull request by chmelevskij**

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/Beepers.js
Comment (haslinghuis) [inline src/js/Beepers.js:1]: This should go in utils - but out of scope of this PR.

**Review by haslinghuis — APPROVED** [review]

File: src/js/Beepers.js
Comment (chmelevskij) [reply src/js/Beepers.js:1]: Definitely. Once modules are done I will go and do another pass. This highlighted quite a few circular dependencies which can we solved with simple re-org of files/modules.

**Comment by haslinghuis** [comment]
@chmelevskij Found an error when using flash firmware button:
image

Think this PR needs to include serial.js?

File: src/js/msp/MSPConnector.js
Comment (chmelevskij) [inline src/js/msp/MSPConnector.js:2]: @haslinghuis this should fix firmware flasher issue.

Comment (haslinghuis) [reply src/js/msp/MSPConnector.js:2]: Thanks.

**Review by haslinghuis — APPROVED** [review]

**Comment by haslinghuis** [comment]
@chmelevskij please rebase

**Comment by haslinghuis** [comment]
Tested on Android. Connection works (Except virtual connection off course but we'd discussed that)

**Comment by haslinghuis** [comment]
Thanks, tested again and confirm, virtual connection now works in Android again.
Ready to merge (after rebase :)

**Review by blckmn — APPROVED** [review]

**Pull request merged**
